src/article.py

```diff
@@ -127,32 +131,33 @@ def bib_key(self):
         For title: Remove commas as well as all common propositions and
         articles (i.e. 'on', 'in', 'a', 'the', ...); then take the first
         three words.
+        The arXiv identifier is added at the end to guarantee uniqueness.
         """
         # key for authors
         authors_key = self.authors_contracted
         # key for title
         # remove most common propositions and articles
         to_remove = ["a", "and", "in", "of", "on", "or", "the", "for"]
         title_key = delete_prepositions(self.title, to_remove, case_sensitive=False)
-        # print(title_key)
         # remove characters which are not allowed/unnecessary in bibtex key
         remove_chars = re.compile(r'[\\{},~#%:"]')
         title_key = re.sub(remove_chars, "", title_key)
         title_key_split = title_key.split()
         title_key = "".join(t.capitalize() for t in title_key_split[:3])
-        # TODO: add arxiv identifier only to make key unique; better way?
+
         return authors_key + "-" + title_key + "-" + self.ax_id
 
     def bib(self):
         """Create a bibtex entry for the given article using
         bib_key and bib_tile.
         """
         article_key = self.bib_key()
-        url = "https://arxiv.org/abs/{}".format(self.ax_id)
+        url = f"https://arxiv.org/abs/{self.ax_id}"
         title = bib_title(self.title)
         bib_entry = (
```

Comment:
This imho should be a function:

```
def render_bibentry(bibentry: BibEntry) -> str:
   ...
```
With some nice unit tests.

Reply:
Totally agree, thanks! As mentioned above, I am working on a redesign of the Article class and will add a class (or classes) to deal with the bib entries.